### PR TITLE
Remove one or multiple people in a house

### PR DESCRIPTION
**Pull request by Giang-github**
Description

- In this edit,  I have added javadoc for features
- Edit the delete function as required
- Added a function that displays a list of all added people

### CODE REVIEW

File: src/main/java/org/resitrack/Main.java [hunk 1]
```diff
@@ -60,16 +59,51 @@ public static void managePerson(PersonController personController, Scanner scann
...
+                        System.out.print("Enter Name Person want to delete: ");
+                        String name = scanner.nextLine();
+                        if(personController.deletePeopleByName(name)){
+                            System.out.println("Delete success !");
+                        }else {
+                            System.out.println("Delete false !");
+                        }
+                    }else {
+                        System.out.println("Not found ");
+                    }
```
Comment (ntananh) [inline src/main/java/org/resitrack/Main.java]: I think we should use switch case for code more clean

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 2]
```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
...
 
+    /**
+     * This function is used to delete one or many people
+     *
+     * @param name
+     * @return true if delete process success and vice versa
+     */
+    public boolean deletePeopleByName(String name) {
+        List<Person> peopleByName = findPersonByName(name);
+        if (peopleByName.size() < 0) {
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]:
```suggestion
        if (peopleByName.size() == 0) {
```
the logic here should check for == 0

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 3]
```diff
@@ -4,42 +4,85 @@
...
+        if (personById == null) {
+            return false;
+        }
+        return deletePerson(personById);
+    }
+
+    /**
+     * This function is used to search one or many people
+     */
+
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: remove this empty line

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 4]
```diff
@@ -4,42 +4,85 @@
...
+        System.out.println("---------------------------------------------------------------------");
     }
 
+    /**
+     * This function is used to delete one person
+     *
+     * @param id
+     * @return true if delete process success and vice versa
+     */
+
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: remove this empty line

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 5]
```diff
@@ -4,42 +4,85 @@
...
+                    person.getDob().toString()
+            );
         }
+        System.out.println("---------------------------------------------------------------------");
     }
 
+    /**
+     * This function is used to delete one person
+     *
+     * @param id
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please explain where is this param come from, what is value of it

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 6]
```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
         }
     }
 
+    /**
+     * This function is used to delete one or many people
+     *
+     * @param name
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please explain where is this param come from, what is value of it

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 7]
```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
...
+        if (peopleByName.size() < 0) {
+            return false;
+        }
+        return deletePeople(peopleByName);
+    }
+
+    /**
+     * This function is used to show basic information of one person
+     *
+     * @param person
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please explain where is this param come from, what is value of it

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 8]
```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
...
         System.out.println("Id: " + person.getId());
         System.out.println("Name: " + person.getName());
         System.out.println("Gender: " + person.getGender());
         System.out.println("Dob: " + person.getDob());
     }
 
+    /**
+     * This function is used to delete many people
+     *
+     * @param people
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please explain where is this param come from, what is value of it

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 9]
```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
...
+     * @return true if delete people success and vice versa
+     */
+
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please remove empty line

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 10]
```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
...
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This function is used to restore list people if delete process false
+     *
+     * @param deletePeople
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please explain where is this param come from, what is value of it

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 11]
```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
...
+     */
+    private void restorePeople(List<Person> deletePeople) {
+        people.addAll(deletePeople);
+    }
+
+    /**
+     * This function is used to delete one person
+     *
+     * @param person
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please explain where is this param come from, what is value of it

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 12]
```diff
@@ -105,6 +241,13 @@ private String generatePersonId() {
         return "P" + count.getAndIncrement();
     }
 
+    /**
+     * This function is used to add new person
+     *
+     * @param person
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: please explain where is this param come from, what is value of it

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 13]
```diff
@@ -105,6 +241,13 @@ private String generatePersonId() {
...
+     * @return true if add person success and vice versa
+     */
+
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: pls remove this empty line

File: src/main/java/org/resitrack/Main.java [hunk 14]
```diff
@@ -60,16 +59,51 @@ public static void managePerson(PersonController personController, Scanner scann
...
             switch (choice) {
-                case "1":
-                    personController.addNewPerson();
+                case ADD_PERSON:
+                    System.out.println("Adding new person");
+                    boolean personBeAdded = personController.addNewPerson();
+                    if (personBeAdded)
+                        System.out.println("Add success !");
+                    else
+                        System.out.println("Add false !");
```
Comment (ntananh) [inline src/main/java/org/resitrack/Main.java]: I think we should create a  String variable to hold value of log string and we can make the value reassign depend on boolean value.
 suggestion
image

File: src/main/java/org/resitrack/Main.java [hunk 15]
```diff
@@ -60,16 +59,51 @@ public static void managePerson(PersonController personController, Scanner scann
...
+                    personController.showAll();
+                    break;
+                case SEARCH_PERSON:
                     personController.searchPeopleByName();
                     break;
+                case DELETE_PERSON:
+                    System.out.println("Delete function: ");
+                    System.out.println("1. Delete person by ID");
+                    System.out.println("2. Delete people by name");
+                    System.out.println("> Enter your choose: ");
```
Comment (ntananh) [inline src/main/java/org/resitrack/Main.java]:
```suggestion
                    System.out.println("> Enter your choice: ");
```

File: src/main/java/org/resitrack/Main.java [hunk 16]
```diff
@@ -60,16 +59,51 @@ public static void managePerson(PersonController personController, Scanner scann
...
+                    String choose = scanner.nextLine();
+                    if (choose.equals("1")) {
+                        System.out.print("Enter IdPerson want to delete: ");
```
Comment (ntananh) [inline src/main/java/org/resitrack/Main.java]:
```suggestion
                        System.out.print("Enter person id want to delete: ");
```

File: src/main/java/org/resitrack/controller/HouseController.java [hunk 17]
```diff
@@ -1,4 +1,8 @@
 package org.resitrack.controller;
 
 public class HouseController {
+    PersonController personController;
+    public void addMemberToHouse(){
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/HouseController.java]: What is the purpose of this function, I can't see it being used.

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 18]
```diff
@@ -4,42 +4,85 @@
...
 import java.util.concurrent.atomic.AtomicInteger;
 
+import static org.resitrack.util.CommonUtil.INITIAL_ID_VALUE;
+
 
 public class PersonController {
     private static final AtomicInteger count = new AtomicInteger(INITIAL_ID_VALUE);
-    private final List<Person> people;
-    private final Scanner scanner;
+    private List<Person> people;
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java:14]: please add javadoc for attributes

File: src/main/java/org/resitrack/controller/PersonController.java [hunk 19]
```diff
@@ -70,13 +176,43 @@ private List<Person> findPersonByName(String name) {
...
+        }
+        return null;
+    }
+
+    /**
+     * This function is used to create one person
+     *
+     * @param gender
+     * @param name
+     * @return {@link Person} Person created
```
Comment (ntananh) [inline src/main/java/org/resitrack/controller/PersonController.java]: explain

**Comment by ntananh** [comment]
Hi Giang,

I see you trying to delete a person object, not remove a person from a house. The current logic is good now.

But i have some points want to discuss with you. If we want to remove a person from a house, we should get the house id to list all people inside this house. After that let user enter the person id and remove the relationship between that person and house.

Remove a person or multiple people out of a house not mean these people died or something cannot exist, so complete remove person from app data seem not a correct logic.

**Comment by ntananh** [comment]
Further more, you also add more functional in this pull request, I think some functions is related to other issue not this issue. Each pull request we want to resolve a issue and if you add more functions not for this issue it will be complicate to review an manage other issue. Other functions not related to this issue is " add new person..."

**Review by ntananh — COMMENTED** [review]
First review is done, feel free to take a look

File: src/main/java/org/resitrack/Main.java (hunk 1 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/Main.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 2 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 3 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 4 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 5 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 6 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Đã update

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 7 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 8 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 9 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 10 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 11 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 12 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 13 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: Updated

File: src/main/java/org/resitrack/Main.java (hunk 14 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/Main.java]: I applied

File: src/main/java/org/resitrack/Main.java (hunk 15 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/Main.java]: I applied

File: src/main/java/org/resitrack/Main.java (hunk 16 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/Main.java]: I applied

File: src/main/java/org/resitrack/controller/HouseController.java (hunk 17 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/HouseController.java]: I added too much, I deleted it

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 18 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java:14]: I added

File: src/main/java/org/resitrack/controller/PersonController.java (hunk 19 above)
Comment (Giang-github) [reply src/main/java/org/resitrack/controller/PersonController.java]: I explained

**Comment by Giang-github** [comment]
I also remove functions that not relate to this issue

**Comment by n-tananh** [comment]
#25

**Comment by ntananh** [comment]
Hi @Giang-github ,

we have some merge conflict, could you please resolve it first

**Pull request closed without merging**